extensions/avro/deployment/src/main/java/org/apache/camel/quarkus/component/avro/deployment/AvroProcessor.java

```diff
@@ -0,0 +1,124 @@
+        items.add(new ReflectiveClassBuildItem(false, false, GenericContainer.class));
+        return items;
+    }
+
+    @BuildStep
+    void additionalBeanClasses(BuildProducer<AdditionalBeanBuildItem> additionalBeanProducer) {
+        additionalBeanProducer.produce(new AdditionalBeanBuildItem(AvroDataFormatProducer.class));
+    }
+
+    @BuildStep
+    AnnotationsTransformerBuildItem markFieldsAnnotatedWithBuildTimeAvroDataFormatAsInjectable() {
+        return new AnnotationsTransformerBuildItem(new AnnotationsTransformer() {
+
+            public boolean appliesTo(org.jboss.jandex.AnnotationTarget.Kind kind) {
+                return kind == org.jboss.jandex.AnnotationTarget.Kind.FIELD;
+            }
+
+            @Override
+            public void transform(TransformationContext ctx) {
+                FieldInfo fieldInfo = ctx.getTarget().asField();
+                if (fieldInfo.annotation(BUILD_TIME_AVRO_DATAFORMAT_ANNOTATION) != null) {
+                    ctx.transform().add(Inject.class).done();
+                }
+            }
+        });
+    }
+
+    @BuildStep
+    void overrideAvroSchemaRegistrySerialization(BuildProducer<ObjectSubstitutionBuildItem> substitutions) {
+        Holder<AvroSchemaRegistry, byte[]> holder = new Holder(AvroSchemaRegistry.class, byte[].class,
+                AvroSchemaRegistrySubstitution.class);
+        substitutions.produce(new ObjectSubstitutionBuildItem(holder));
+    }
+
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    void recordAvroSchemaRegistryInitialization(BeanArchiveIndexBuildItem beanArchiveIndex,
+            BeanContainerBuildItem beanContainer, AvroRecorder avroRecorder) {
+        AvroSchemaRegistry avroSchemaRegistry = new AvroSchemaRegistry();
+        IndexView index = beanArchiveIndex.getIndex();
+        for (AnnotationInstance annotation : index.getAnnotations(BUILD_TIME_AVRO_DATAFORMAT_ANNOTATION)) {
+            String schemaResourceName = annotation.value().asString();
+            FieldInfo fieldInfo = annotation.target().asField();
+            String injectedFieldId = fieldInfo.declaringClass().name() + "." + fieldInfo.name();
+            try (InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream(schemaResourceName)) {
+                Schema avroSchema = new Schema.Parser().parse(is);
+                avroSchemaRegistry.put(injectedFieldId, avroSchema);
+                LOG.debug("Parsed the avro schema at build time from resource named " + schemaResourceName);
+            } catch (SchemaParseException | IOException ex) {
+                final String message = "An issue occured while parsing schema resource on field " + injectedFieldId;
+                throw new BuildTimeAvroSchemaParseException(message, ex);
+            }
+        }
+        avroRecorder.recordDataFormatProducerRegistryInitialization(beanContainer.getValue(), avroSchemaRegistry);
```

Comment:
I wonder whether it would not be safer to not to rely on HashMap serialization and instead create the HashMap in the recorder and pass the key-schema pairs one by one there. We'd thus still depend on Schema serialization but the amount of possible breakage would be a bit smaller.

Reply:
Schema serialization would fail due to Quarkus issue [#9446]

Reply:
I mean something like this: https://github.com/ppalaga/camel-quarkus/commit/5d03784f8907b045188f46ec9aaeebee20b9018a#diff-4a5870489e7ebf2120398aaf3099ee6aL106-R115

Reply:
ok, linked to discussion above.